### PR TITLE
treat warnings as error [posix]

### PR DESCRIPTION
**Pull request by guliashvili**
Treating warnings as errors in posix 

- makes osquery coding style better. 
- In our current setup, it is hard to check if PR is causing additional warnings/problems and reviewer time was spent on pointing out pedantic things.
- makes review faster (Less ping pong between the reviewer and the author)

I had to disable single macOS warning. Discussed with @obelisk . Looks like we can not easily replace pthread_setugid_np and also it should not be removed any time soon. So, it is okay to ignore this warning.

### CODE REVIEW

**Comment by akindyakov** [comment]
This PR gonna fix issue #4635 . Let me mention it here.

File: osquery/core/system.cpp [hunk 1]
```diff
@@ -564,11 +568,12 @@ DropPrivileges::~DropPrivileges() {
 #endif
 
 Status setThreadName(const std::string& name) {
-  int return_code;
 #if defined(__APPLE__)
-  return_code = pthread_setname_np(name.c_str());
+  int return_code = pthread_setname_np(name.c_str());
+  return Status(return_code == 0 ? 0 : 1);
```
Comment (akindyakov) [inline osquery/core/system.cpp]: Would you like to make it explicit? Smth like:
```c++
return return_code == 0 ? Status::success() : Status::failure();
```

Comment (guliashvili) [reply osquery/core/system.cpp]: Sure. Thanks for pointing out

**Comment by fmanco** [comment]
I would be happier if this PR was split 😄

**Pull request closed without merging**